### PR TITLE
Lesson 1

### PR DESCRIPTION
**Pull request by paaaaaaul**

### CODE REVIEW

File: src/Lesson1/ArrayRotation.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+1 2 6 5 6
+*
+* */
+
+package Lesson1;
+import java.util.Scanner;
+
+public class ArrayRotation {
+
```
Comment (BondarenkoJek) [inline src/Lesson1/ArrayRotation.java]: delete line

File: src/Lesson1/ArrayRotation.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        for (int i = 0; i < size; i++) { // в цикле заполняем массив
+            arr[i] = scanner.nextInt();
+        }
+        moveToEnd(arr, size, elToMove); //передаем параметры в метод, который будет переносить элементы массива
+
+        for (int i = 0; i < n; ++i) {
+            System.out.print(arr[i] + " "); //выводим результат в консоль
+        }
+    }
+}
```
Comment (BondarenkoJek) [inline src/Lesson1/ArrayRotation.java]: should add an empty line after bacets

File: src/Lesson1/Ranges/RangeDefiner.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package Lesson1.Ranges;
+
+import java.util.Scanner;
+
+public class RangeDefiner {
+
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/RangeDefiner.java]: delete line

File: src/Lesson1/Ranges/RangeDefiner.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            prev = cur; // записываем прев и ждем следующую итерацию
+        }
+        if (count != 1) {
+            builder.append(" ").append(prev);
+        }
+        builder.append("]");
+        System.out.println(builder.toString());
+        return builder.toString();
+    }
+}
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/RangeDefiner.java]: add an empty line

File: src/Lesson1/Ranges/Ranges.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/Ranges.java]: delete line

File: src/Lesson1/Ranges/Ranges.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+Output
+[1 3][5][8 10][13 16]
+
+
+*
+* */
+
+package Lesson1.Ranges;
+
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/Ranges.java]: delete line

File: src/Lesson1/Ranges/Ranges.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/Ranges.java]: delete line

File: src/Lesson1/Ranges/Ranges.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+public class Ranges {
+
+    public static void main(String[] args) {
+        RangeDefiner.defineTheRange();
+        //
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/Ranges.java]: delete line

File: src/Lesson1/Ranges/Ranges.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+
+*
+* */
+
+package Lesson1.Ranges;
+
+
+public class Ranges {
+
```
Comment (BondarenkoJek) [inline src/Lesson1/Ranges/Ranges.java]: delete line

File: src/Lesson1/RestoreArray/RestoreArray.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+8
+1 2 -1 4 7 6 -2 4
+
+Output
+1 2 3 4 7 6 5 4
+
+* */
+
+package Lesson1.RestoreArray;
+
```
Comment (BondarenkoJek) [inline src/Lesson1/RestoreArray/RestoreArray.java]: delete line

File: src/Lesson1/RestoreArray/RestoreArray.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+public class RestoreArray {
+
```
Comment (BondarenkoJek) [inline src/Lesson1/RestoreArray/RestoreArray.java]: delete line

File: src/Lesson1/RestoreArray/RestoreArray.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    public static void main(String[] args) {
+       TheRealRestore.restoreTheArray();
+    }
+}
```
Comment (BondarenkoJek) [inline src/Lesson1/RestoreArray/RestoreArray.java]: add an empty line line

File: src/Lesson1/RestoreArray/TheRealRestore.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
+package Lesson1.RestoreArray;
+
+import java.util.Scanner;
+
+public class TheRealRestore {
+
```
Comment (BondarenkoJek) [inline src/Lesson1/RestoreArray/TheRealRestore.java]: delete line

File: src/Lesson1/RestoreArray/TheRealRestore.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                    monotoneArray[i] = cur;
+                    System.out.print(" ");
+                }
+            }
+            System.out.print(monotoneArray[i] + " ");
+        }
+        System.out.println("\n Restoring is finished");
+        return "finished";
+    }
+
```
Comment (BondarenkoJek) [inline src/Lesson1/RestoreArray/TheRealRestore.java]: delete line

**Review by BondarenkoJek — CHANGES_REQUESTED** [review]

File: src/Lesson1/ArrayRotation.java [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+        for (int i = 0; i < size; i++) { // в цикле заполняем массив
+            arr[i] = scanner.nextInt();
+        }
+        moveToEnd(arr, size, elToMove); //передаем параметры в метод, который будет переносить элементы массива
+
+        for (int i = 0; i < n; ++i) {
+            System.out.print(arr[i] + " "); //выводим результат в консоль
+        }
+
```
Comment (BondarenkoJek) [inline src/Lesson1/ArrayRotation.java:46]: delete line

File: src/Lesson1/RangeDefiner.java [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+        builder.append("]");
+        System.out.println(builder.toString());
+        return builder.toString();
+
+    }
+
+    public static void main(String[] args) {
+        RangeDefiner.defineTheRange();
+    }
+}
```
Comment (BondarenkoJek) [inline src/Lesson1/RangeDefiner.java]: should add an empty line after bacets

**Review by BondarenkoJek — CHANGES_REQUESTED** [review]

File: src/Lesson1/RangeDefiner.java (hunk 16 above)
Comment (paaaaaaul) [reply src/Lesson1/RangeDefiner.java]: after 50th line?

Comment (BondarenkoJek) [reply src/Lesson1/RangeDefiner.java]: 51

**Review by BondarenkoJek — APPROVED** [review]
LGTM

**Pull request merged**